avro-serializer/src/main/java/io/confluent/kafka/serializers/AbstractKafkaAvroDeserializer.java

```diff
@@ -157,23 +152,10 @@ protected Object deserialize(boolean includeSchemaAndVersion, String topic, Bool
 
         Integer version = schemaVersion(topic, isKey, id, subject, schema, result);
 
-        if (schema.getType() == Schema.Type.UNION) {
-          // Can't set additional properties on a union schema since it's just a list, so set it
-          // on the first non-null entry
-          for (Schema memberSchema : schema.getTypes()) {
-            if (memberSchema.getType() != Schema.Type.NULL) {
-              memberSchema.addProp(SCHEMA_REGISTRY_SCHEMA_VERSION_PROP,
-                                   JsonNodeFactory.instance.numberNode(version));
-              break;
-            }
-          }
-        } else {
-          setVersionProp(schema, version);
-        }
         if (schema.getType().equals(Schema.Type.RECORD)) {
-          return result;
+          return new GenericContainerWithVersion((GenericContainer) result, version);
```

Comment:
@rayokota I believe this is potentially an incompatible change. If there are connectors that are looking for the version property, they could potentially break with this change?